dotnet/src/SemanticKernel.Core/Kernel.cs

```diff
@@ -357,10 +354,10 @@ public void Dispose()
         // is invoked manually without a context and without a way to find other functions.
         func.SetDefaultFunctionCollection(this.Functions);
 
-        func.SetAIConfiguration(functionConfig.PromptTemplateConfig.Completion);
+        func.SetAIConfiguration(functionConfig.PromptTemplateConfig.Models.FirstOrDefault<AIRequestSettings>());
 
         // Note: the service is instantiated using the kernel configuration state when the function is invoked
-        func.SetAIService(() => this.GetService<ITextCompletion>(functionConfig.PromptTemplateConfig.Completion?.ServiceId ?? null));
+        func.SetAIService(() => this.GetService<ITextCompletion>(functionConfig.PromptTemplateConfig.Models.FirstOrDefault<AIRequestSettings>()?.ServiceId ?? null));
```

Comment:
So many places with this `Models.FirstOrDefault<AIRequestSettings>()`, might worth moving this to a method for consistency

Reply:
Ha, I already created the method I just didn't update all of the code to use it.